consoles/sshVirtsh.pm

```diff
@@ -522,7 +522,7 @@ __END"
     # define the new domain
     $self->run_cmd("virsh $remote_vmm define $xmlfilename") && die "virsh define failed";
     if ($self->vmm_family eq 'vmware') {
-        my $vmx = sprintf('/vmfs/volumes/datastore1/openQA/%s.vmx', $self->name);
+        my $vmx = sprintf('/vmfs/volumes/' . $bmwqemu::vars{VMWARE_DATASTORE} . '/openQA/%s.vmx', $self->name);
```

Comment:
It is nicer to just rely on `sprintf` and still specify `datastore1` as a default:
```suggestion
        my $vmx = sprintf('/vmfs/volumes/%s/openQA/%s.vmx', $bmwqemu::vars{VMWARE_DATASTORE} // 'datastore1', $self->name);
```
(or at least use interpolation)

If you think `datastore1` is too specific to use as a default and there should be no default then you should make it rather throw an exception that the mandatory variable `VMWARE_DATASTORE` is missing (instead of having Perl emit a warning in that case).

Reply:
Looks like the variable `VMWARE_DATASTORE` is already used somewhere else defaulting to an empty string. Not sure whether that makes sense, though. If you make it defaulting to `datastore1` it would make sense to also update other places accordingly and specify the default in `doc/backend_vars.asciidoc`.

Reply:
Thanks for your review and comments.
Actually, the default datastore name `datastore1` is nonexistent in our vmware server which caused the job failure. This code change is for the case that `datastore1` is not specified as the default value.
Agree your suggested change, will update it and specify the default value in `doc/backend_vars.asciidoc`.